compiler/src/dotty/tools/dotc/typer/ErrorReporting.scala

```diff
@@ -16,7 +16,7 @@ object ErrorReporting {
   import tpd._
 
   def errorTree(tree: untpd.Tree, msg: => Message)(implicit ctx: Context): tpd.Tree =
-    tree withType errorType(msg, tree.pos)
+    tree withType errorType(msg, tree.pos.focus)
```

Comment:
I'm not so sure anymore that this is a good idea, since it can lead to worse error messages

Reply:
We only desired this effect for one call site: https://github.com/lampepfl/dotty/pull/4385/commits/0f2d1a3aa1d5d2c54f7a8664de3063577e40e871. So closing this PR.

Medium-term, I think you'd want positions inside `TypeError`, but that shouldn't block #4385.